### PR TITLE
Add Wikisource dataset

### LINKED ISSUE #3399: Add Wikisource dataset
## Adding a Dataset
- **Name:** *wikisource*
- **Description:** *short description of the dataset (or link to social media or blog post)*
- **Paper:** *link to the dataset paper if available*
- **Data:** *link to the Github repository or current dataset location*
- **Motivation:** Additional high quality textual data, besides Wikipedia.

Add loading script as "canonical" dataset (as it is the case for ""wikipedia").

Instructions to add a new dataset can be found [here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md).

CC: @geohci, @yjernite

### PR DESCRIPTION
**Pull request by albertvillanova**
Add loading script for Wikisource dataset.

Fix #3399.

CC: @geohci, @yjernite

### CODE REVIEW

File: datasets/wikisource/wikisource.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+"""
+
+_HOMEPAGE = "https://dumps.wikimedia.org"
+
+_LICENSE = "Creative Commons Attribution-ShareAlike 3.0 Unported"
+
+_BASE_URL_TMPL = "https://dumps.wikimedia.org/{lang}wikisource/{date}/"
+_INFO_FILE = "dumpstatus.json"
+
+WIKISOURCE_LANGUAGES = []
```
Comment (geohci) [inline datasets/wikisource/wikisource.py:49]: FYI if you're looking to build a static list of language codes, these statistics might be of help for guiding that choice: https://meta.wikimedia.org/wiki/Wikisource#List_of_Wikisources (where the `good` column is the one to pay attention to as # of content pages)

File: datasets/wikisource/wikisource.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            """Cleans raw wikicode to extract text."""
+            id_, title, raw_content = inputs
+            try:
+                text = _parse_and_clean_wikicode(raw_content, parser=mwparserfromhell)
+            except (mwparserfromhell.parser.ParserError) as e:
+                beam.metrics.Metrics.counter(language, "parser-error").inc()
+                logger.error("mwparserfromhell ParseError: %s", e)
+                return
+
+            if not text:
```
Comment (geohci) [inline datasets/wikisource/wikisource.py:185]: Given that it's not uncommon for Wikisource pages to have a small amount of metadata (e.g., a few categories) but not contain the transcribed text of the document yet, we probably want to also add a basic length filter. In my exploration, a minimum of 100 characters seems to be a good trade-off between removing many useless documents while retaining those that do have content. So: `if not text or len(text) < 100:` (and obviously could move the 100 up to a config variable to make it more salient of a choice.

**Review by geohci — COMMENTED** [review]
Left a few comments but I think mainly we'll just want to make sure that whatever changes are accepted around the wikipedia dataset script are mirrored here

File: datasets/wikisource/wikisource.py (hunk 2 above)
Comment (albertvillanova) [reply datasets/wikisource/wikisource.py]: Done.

File: datasets/wikisource/wikisource.py (hunk 1 above)
Comment (albertvillanova) [reply datasets/wikisource/wikisource.py]: DONE @geohci.

**Pull request closed without merging**